checker/src/main/java/org/checkerframework/checker/nullness/NullnessNoInitVisitor.java

```diff
@@ -776,9 +788,38 @@ public void visitSwitchExpression17(Tree switchExprTree) {
             checkForNullability(
                     SwitchExpressionUtils.getExpression(switchExprTree), SWITCHING_NULLABLE);
         }
+        if (lintredundantNullComparison) {
+            ExpressionTree expression = SwitchExpressionUtils.getExpression(switchExprTree);
+            List<? extends CaseTree> cases = SwitchExpressionUtils.getCases(switchExprTree);
+            checkSwitchNullRedundant(expression, cases);
+        }
         super.visitSwitchExpression17(switchExprTree);
     }
 
+    /**
+     * Reports an error if the switch expression is @NonNull and the case expression is null
+     * literal.
+     *
+     * @param expression the switch expression
+     * @param cases the case expressions
+     */
+    private void checkSwitchNullRedundant(
+            ExpressionTree expression, List<? extends CaseTree> cases) {
+        AnnotatedTypeMirror switchType = atypeFactory.getAnnotatedType(expression);
+        for (CaseTree caseTree : cases) {
+            List<? extends ExpressionTree> caseExpressions =
+                    TreeUtilsAfterJava11.CaseUtils.getExpressions(caseTree);
+            for (ExpressionTree caseExpression : caseExpressions) {
+                if (caseExpression != null
```

Comment:
When is this null? Maybe for switch statements only do something for enhanced switch statements?
https://github.com/eisop/checker-framework/blob/d7f9dabf1240e40484ce2b3c1c6e9feecaabe29c/javacutil/src/main/java/org/checkerframework/javacutil/TreeUtils.java#L2447

Suggestion above probably addresses this.

Reply:
I will remove the checks here.